### PR TITLE
⚠️ Update to gophercloud 2.0.0-beta2

### PR DESCRIPTION
**Pull request by dtantsur**
Although the current version is specified as 1.5.1, it's actually
a commit somewhere in the 2.0 branch. This change brings us to
a more specific version.

2.0.0-beta2 introduced a requirement on Context for all calls.

Signed-off-by: Dmitry Tantsur <dtantsur@protonmail.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,11 +1,11 @@
 module github.com/metal3-io/baremetal-operator
 
-go 1.21
+go 1.21.6
```
Comment (kashifest) [inline go.mod:3]: Any specific reason to pin it to patch version?

Comment (dtantsur) [reply go.mod:3]: This is apparently the new way Go does it starting with 1.21. I haven't done this change - `go mod tidy` does it (based on Gophercloud's go.mod, I believe).

Comment (kashifest) [reply go.mod:3]: Interesting , I didnt face it when I bumped it to go 1.21, I have to check it again.

**Review by kashifest — COMMENTED** [review]
/approve

**Pull request merged**